daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -937,13 +952,41 @@ private[lf] object SBuiltin {
     }
   }
 
+  /** $checkTemplate[T] :: Unit -> bool */
+  private[speedy] final case class SBCheckTemplate(templateId: TypeConName) extends SBuiltin(1) {
```

Comment:
Good question. In a sense, this really is an `SBuiltinPure`, however, we need access to `machine` (which `SBuiltinPure` does not allow) so that we can query `compiledPackages`.

In an ideal world, `SBuiltinPure` should allow access to an immutable version of `machine`, but that is not currently possible.

Reply:
The current _operational_ definition of `SBuiltinPure`, is anything which can define:
```
  private[speedy] def executePure(args: util.ArrayList[SValue]): SValue
```
perhaps we can also pass the `Machine` here, as Carl suggests :thinking:

Reply:
For the moment, I've added in the following todo:
```
    // TODO: when we have a concept of immutable machine or compiled packages, make this a pure builtin
```

Reply:
My bad, it should not be a SBuiltinPure.

Reply:
No problem. I'll remove these comments.

Reply:
> I'd suggest not passing the machine (or even `compiledPackages`) at this point in time - they are both mutable.

:+1: